# **Galaxy training report**

---
### **Introduction**

[Galaxy](https://galaxyproject.org/community/) is an online platform developed by a community of scientists. This platform is distributed in different regions by presence of [servers](https://galaxyproject.org/usegalaxy/). These servers are accessible virtually and contain various tools which aid in the analysis of different types of data. From galaxy one can access training materials from the galaxy training network (GTN) [page](https://training.galaxyproject.org/). These page consists  of a wide range of tutorials on how to use the inbuilt tools in galaxy to carry out data analysis. From 15th to 19th February 2021, I participated virtually in the GTN Smörgåsbord: [A Global Course](https://training.galaxyproject.org/training-material/smorgasbord.html). This workshop comprised of different GTN tutorials exposing one to a wide range topics during the entire week.

---

### **Objectives**
1. To familiarize with galaxy platform and learn data analysis using galaxy.
2. To understand NGS analysis using galaxy tools.
3. To determine the importance of using certain tools for your data analysis over others.

---

### **Skills learnt**

1. Carrying out Next-Generation analysis using galaxy tools.
   Once one receives sequenced data from a sequencer, there are a set of steps that are carried out in the analysis of the data. Galaxy platforms comprise of a set of tools that enables step-wise analysis of the data. Quality control is an essential first step to any sequenced data so that the downstream analysis of the data is not affected by the state of the data. Galaxy has a FastQc tool that checks the quality of the sequenced files. It assesses the data based on various parameters such as sequence quality score and per base sequence quality. Depending on the output of the fastQc tool, other tools come in handy when quality correction is necessary. These tools include Cutadapt and Filter tool. Cutadapt trims adapter sequences as well as low-quality score regions from the FastQ data. Filter tool on the there hand, removes N basecalls in the sequence, reads with short sequence length, and those with low mean quality scores. These tools aid in improving the quality of the sequenced data. Additionally, the MultiQC tool is useful in the aggregation of different output files from the same tool that was used. This makes it easier to compare the results at once.
   Sequencing tools such as Illumina produce short reads of sequences whose genomic context and their location on the genome. Thus alignment of the short reads to a reference genome is done so as to know their specific location on the genome and insights about the reads. There are various tools offered by galaxy for mapping reads like Bowtie2, BWA-MEM, and many more. Mapping reads is important to know any potential errors in the reads and what to do about them. Mapping tools generate a Binary Alignment Map of the reads to a sequence genome, their location, and the quality of mapping. The output of the BAM files is then visualized using visualizers including (but not limited to) JBrowse and Integrative Genome Viewer (IGV). Visualization enables an interactive interface to view aligned reads to the genome and note any differences between the two. In galaxy there are very many tools and algorithms for mapping, it is dependent on one's data the outcome one desires.
    Unlike mapping of reads to a reference genome, one may decide to assemble the short fragments back to an original DNA sequence. This process is known as De novo assembly. Although theoretically, this process sounds simple it is quite difficult. Flye is an example of a tool for assembling reads from a nanopore sequencer. Nanopore sequencers generate long reads which might have errors during base calling, hence short reads from Illumina sequencers are used to polish and correct errors in the assembled reads. Pilon is an example of a tool that polishes long reads using the short reads from an Illumina sequencer. Once errors are removed, annotation is done of the assembled reads using tools like Prokka and Geseq. Annotation of the reads is then followed by visualization using preferred visualizers.
	
2. RNA-Seq analysis using Galaxy and Rstudio.
   RNA sequencing analyzes cellular transcriptome ( RNA molecules from a cell or a tissue) to find out which genes are upregulated or downregulated. RNA-seq is done mainly to identify novel genes and even differentially expressed genes or molecular pathways in two or more biological conditions. The workflow of RNA-seq is quite similar to an NGS analysis, in that quality control is needed to determine the quality of your reads and whether improvement of the quality is required. After quality control, mapping is carried out to identify the location where the reads originated from. For RNA sequences it is somehow different from the mapping of DNA to a reference genome due to the presence of introns and splicing of the RNA therefore special tools such as RNASTAR are available for mapping RNA sequences. Visualization of the mapped reads using IGV tool which also enables one to view spliced regions using Sashimi plots. Before figuring out the differential expressed genes, it is necessary to count the reads occurring in annotated genes.
   An important step in RNA-seq workflow is to investigate for any differentially expressed genes. This can be done using the DESEq2 tool. DeSeq2 tool does this by estimating the presence of biological variance and significance of expression differences between any two conditions. Factors with several levels are incorporated in the analysis to describe known sources of variation. Differentially expressed genes are visualized using heatmaps. Gene ontology of the identified differentially expressed genes provides the functionality of the genes and if needed KEGG pathways where the genes are involved could also be investigated. Using Rstudio, visualization and data manipulation of the analyzed RNA-seq data can also be done. Rstudio has several packages such as ggplot2 which makes it easier to produce customized results for your work which couldn't be so when using galaxy tools. With ggplot2 package for instance one can set different parameters and get scatter plots, boxplots, trendlines, and many more. It generally depends on your desired graphical presentation of data. 




